commit 6fcbb46dda912c2ca407076e45b5c8cefdea90c6
Author: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
Date:   Sun Apr 11 04:43:17 2010 +0900

    ping6: do not allow too large packet size by -s option.
    
    Debian BTS #166845
    
    Signed-off-by: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>

Resolves: rhbz#1010296

diff --git a/ping_common.c b/ping_common.c
index 0a6d869..1de34b3 100644
--- a/ping_common.c
+++ b/ping_common.c
@@ -202,6 +202,11 @@ void common_options(int ch)
 			fprintf(stderr, "ping: illegal negative packet size %d.\n", datalen);
 			exit(2);
 		}
+		if (datalen > maxpacket - 8) {
+			fprintf(stderr, "ping: packet size too large: %d\n",
+				datalen);
+			exit(2);
+		}
 		break;
 	case 'v':
 		options |= F_VERBOSE;
